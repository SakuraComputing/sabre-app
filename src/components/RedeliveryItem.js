import React from 'react';
import { NavLink } from 'react-router-dom';
import moment from 'moment';

const RedeliveryItem = ({ cardId, parcelId, addressNo, latestScan, requestDate}) =>  (
    <tr className="redelivery-item">
        <td>
            <NavLink to={`/redelivery/${cardId}`} activeClassName="is-active">
                {cardId}
            </NavLink>
        </td>
        <td>{parcelId}</td>
        <td>{addressNo}</td>
        <td>{moment(latestScan).format('dddd, MMMM Do YYYY, h:mm:ss a')}</td>
        <td>{moment(requestDate).format('dddd, MMMM Do YYYY, h:mm:ss a')}</td>
    </tr>
);

export default RedeliveryItem;