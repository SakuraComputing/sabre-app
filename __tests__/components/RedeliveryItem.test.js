import React from 'react';
import { shallow } from 'enzyme';
import RedeliveryItem from '../../src/components/RedeliveryItem';

const redeliveryItem = {
  cardId: '999999',
  parcelId: 'JD0002567834521111',
  addressNo: 5,
  latestScan: 1511264704
};

describe('Redelivery Item Component',() => {
    it('should render the form correctly',() => {
        const wrapper = shallow(<RedeliveryItem redeliveryItem={redeliveryItem}/>);
        expect(wrapper).toMatchSnapshot();
    })
});