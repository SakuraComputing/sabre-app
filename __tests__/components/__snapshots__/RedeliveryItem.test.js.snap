// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redelivery Item Component should render the form correctly 1`] = `
<tr
  className="redelivery-item"
>
  <td>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/redelivery/undefined"
    />
  </td>
  <td />
  <td />
  <td>
    Thursday, January 1st 1970, 12:00:00 am
  </td>
  <td>
    Thursday, January 1st 1970, 12:00:00 am
  </td>
</tr>
`;
